### PR TITLE
Fix: Matrix modal

### PR DESCRIPTION
**Pull request by tuul-wq**
- Matrix modal navigation
- Create wallet provider
- Fixed `MatrixLogin` inside `CreateMultisig`
- Fixed smooth animation for CreateWallet modals

### CODE REVIEW

File: src/renderer/components/layout/PrimaryLayout/Wallets/WalletMenu.tsx [hunk 1]
```diff
@@ -72,72 +77,52 @@ const WalletMenu = ({ children, chains, wallets }: PropsWithChildren<Props>) =>
...
-                      />
-                    ))}
-                </ul>
-              </section>
-            )}
-          </Popover.Panel>
-        </Transition>
-      </Popover>
-
-      <WatchOnly
```
Comment (tuul-wq) [inline src/renderer/components/layout/PrimaryLayout/Wallets/WalletMenu.tsx:133]: These 3 modals were executing all the time without render part but with useEffects and creating functions

File: src/renderer/components/modals/CreateMultisigAccount/CreateMultisigAccount.tsx [hunk 2]
```diff
@@ -30,16 +30,17 @@ const enum Step {
...
+  onComplete: () => void;
 };
 
-export const CreateMultisigAccount = ({ isOpen, onClose }: Props) => {
+export const CreateMultisigAccount = ({ isOpen, onClose, onComplete }: Props) => {
   const { t } = useI18n();
   const { matrix, isLoggedIn } = useMatrix();
   const { getWallets } = useWallet();
   const { getAccounts, addAccount, setActiveAccount } = useAccount();
-  const navigate = useNavigate();
```
Comment (tuul-wq) [inline src/renderer/components/modals/CreateMultisigAccount/CreateMultisigAccount.tsx:40]: Modal should not handle navigation, it's his parent work

File: src/renderer/components/modals/CreateMultisigAccount/CreateMultisigAccount.tsx [hunk 3]
```diff
@@ -203,13 +205,18 @@ export const CreateMultisigAccount = ({ isOpen, onClose }: Props) => {
           wallets={signatoryWallets}
           contacts={signatoryContacts}
         />
+
+        <MatrixLoginModal isOpen={!isLoggedIn} zIndex="z-60" onClose={closeMultisigModal} />
```
Comment (tuul-wq) [inline src/renderer/components/modals/CreateMultisigAccount/CreateMultisigAccount.tsx:209]: Headlessui requires modals cascade to close them properly 😟

File: src/renderer/features/matrix/MatrixRouteGuard/ui/MatrixRouteGuard.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { ReactNode } from 'react';
+
+import { useMatrix } from '@renderer/app/providers';
+
+type Props = {
+  children: (isLoggedIn: boolean) => ReactNode;
+};
+export const MatrixRouteGuard = ({ children }: Props) => {
```
Comment (tuul-wq) [inline src/renderer/features/matrix/MatrixRouteGuard/ui/MatrixRouteGuard.tsx:8]: ⚠️ Useless, I will remove it

File: src/renderer/pages/Settings/Overview/components/MatrixAction/MatrixAction.tsx [hunk 5]
```diff
@@ -1,50 +1,46 @@
...
+          className={cnTw(
+            'w-full grid grid-flow-col grid-cols-[auto,1fr,auto] items-center gap-x-2 p-3 rounded-md',
+            'transition hover:shadow-card-shadow focus:shadow-card-shadow',
+          )}
+        >
+          <Icon className="row-span-2" name="matrix" size={36} />
+          <BodyText>{t('settings.overview.matrixLabel')}</BodyText>
+          <HelpText className="text-text-tertiary">{t('settings.overview.matrixDescription')}</HelpText>
 
-      <MatrixModal isOpen={isModalOpen} onClose={toggleModalOpen} />
```
Comment (tuul-wq) [inline src/renderer/pages/Settings/Overview/components/MatrixAction/MatrixAction.tsx:47]: Matrix goes into `Outlet`

File: src/renderer/widgets/CreateWallet/ui/CreateWalletProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import Vault from '@renderer/pages/Onboarding/Vault/Vault';
+import { CreateMultisigAccount } from '@renderer/components/modals';
+import { Paths } from '../../../app/providers/routes/paths';
+
+// TODO: Break down WatchOnly / Vault / CreateMultisig to widgets
+type ModalProps = {
+  onClose: () => void;
+  onComplete: () => void;
+};
+const WalletModals: Record<WalletType, (props: ModalProps) => JSX.Element> = {
```
Comment (tuul-wq) [inline src/renderer/widgets/CreateWallet/ui/CreateWalletProvider.tsx:17]: Every new wallet type will go here

File: src/renderer/widgets/MatrixModal/ui/Credentials.tsx [hunk 7]
```diff
@@ -16,7 +12,6 @@ const Credentials = ({ onLogOut }: Props) => {
     toggleProgress();
     try {
       await matrix.logout();
-      onLogOut();
```
Comment (tuul-wq) [inline src/renderer/widgets/MatrixModal/ui/Credentials.tsx:19]: Now after `logout` we will be redirected to `LoginModal`
Instead of weird modal closure

**Comment by tuul-wq** [comment]
Task linked: [CU-8692t36y5 Matrix login. Can't login while create multisig account](https://app.clickup.com/t/8692t36y5)

**Review by pgolovkin — APPROVED** [review]

**Comment by pgolovkin** [comment]
Should be merged before WalletConnect integration.
@tuul-wq please point Stepan to function areas that has to be checked.

File: src/renderer/widgets/CreateWallet/model/wallet-provider-model.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  navigate(redirectPath);
+});
+
+forward({ from: walletTypeSet, to: $walletType });
+
+sample({
+  clock: [modalClosed, completed],
+  fn: () => null,
+  target: $walletType,
+});
```
Comment (Asmadek) [inline src/renderer/widgets/CreateWallet/model/wallet-provider-model.ts:32]: Maybe better to use `reset` instead of sample with `fn: () => null`?

**Review by Asmadek — APPROVED** [review]

**Pull request merged**